### ISSUE TITLE
foreground normalization

### ISSUE THREAD

**Issue by powder21** [JiahuiYu/generative_inpainting#280]
`yi = tf.nn.conv2d(xi, wi_normed, strides=[1,1,1,1], padding="SAME")

yi = tf.nn.softmax(yi*scale, 3)

yi = tf.nn.conv2d_transpose(yi, wi_center, tf.concat([[1], raw_fs[1:]], axis=0), strides=[1,rate,rate,1]) / 4.
`
The foreground patch are used in these lines, but I don't understand where normalization is canceled. Could u explain it in detail?

**Comment by JiahuiYu** [comment]
It is canceled in softmax. Please have a replacement of two cases in softmax.
